karate-core/src/main/java/com/intuit/karate/StepDefs.java

```diff
@@ -43,309 +33,327 @@
-    public void matchContains(String each, String name, String path, String only, String expected) {
-        MatchType mt = toMatchType(each, only, true);
-        matchNamed(mt, name, path, expected);
-    }
-
-    public void matchNamed(MatchType matchType, String name, String path, String expected) {
-        AssertionResult ar = Script.matchNamed(matchType, name, path, expected, context);
-        handleFailure(ar);
-    }
-
-    @Then("^set ([^\\s]+)( .+)? =$")
-    public void setByPathDocString(String name, String path, String value) {
-        setNamedByPath(name, path, value);
-    }
-
-    @Then("^set ([^\\s]+)( .+)? = (.+)")
-    public void setByPath(String name, String path, String value) {
-        setNamedByPath(name, path, value);
-    }
-
-    public void setNamedByPath(String name, String path, String value) {
-        Script.setValueByPath(name, path, value, context);
-    }
-
-    @Given("^call ([^\\s]+)( .*)?")
-    public final void callAndUpdateVars(String name, String arg) {
-        Script.callAndUpdateVarsIfMapReturned(name, arg, context);
-    }
-
-    private void handleFailure(AssertionResult ar) {
-        if (!ar.pass) {
-            logger.error("{}", ar);
-            throw new KarateException(ar.message);
-        }
-    }
+	private static final Logger logger = LoggerFactory.getLogger(StepDefs.class);
+
+	public StepDefs() { // zero-arg constructor for IDE support
+		this(getFeatureEnv(), null, null);
+	}
+
+	private static ScriptEnv getFeatureEnv() {
+		String cwd = new File("").getAbsoluteFile().getPath();
+		String javaCommand = System.getProperty("sun.java.command");
+		String featurePath = FileUtils.getFeaturePath(javaCommand, cwd);
+		ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+		if (featurePath == null) {
+			File file = new File("");
+			logger.warn("unable to derive feature file path, using: {}", file.getAbsolutePath());
+			return ScriptEnv.init(file, null, classLoader);
+		} else {
+			File file = new File(featurePath);
+			logger.info("ide running: {}", file);
+			return ScriptEnv.init(file.getParentFile(), file.getName(), classLoader);
+		}
+	}
+
+	public StepDefs(ScriptEnv env, ScriptContext parentContext, Map<String, Object> callArg) {
+		context = new ScriptContext(env, parentContext, callArg);
+		request = new HttpRequest();
+	}
+
+	private final ScriptContext context;
+	private HttpRequest request;
+	private HttpResponse response;
+
+	public ScriptContext getContext() {
+		return context;
+	}
+
+	@When("^configure ([^\\s]+) =$")
+	public void configureDocString(String key, String exp) {
+		configure(key, exp);
+	}
+
+	@When("^configure ([^\\s]+) = (.+)")
+	public void configure(String key, String exp) {
+		context.configure(key, exp);
+	}
+
+	@When("^url (.+)")
+	public void url(List<String> paths) {
+		StringBuilder url = new StringBuilder();
+		for (String path : paths) {
+			url.append(Script.eval(path, context).getAsString()).append("/");
```

Comment:
actually this is a trick one and I'm now wondering if we should do this for URL at all (it is fine for path)

for example, I wanted to solve for concatenating the port number etc, and what if the user already concatenated a '/' character